.github/workflows/UnitTests.yml

```diff
@@ -5,90 +5,29 @@ on:
   push:
     branches:
       - master
-
-env:
-  ROBLOSECURITY: ${{ secrets.ROBLOSECURITY || '_|WARNING:-DO-NOT-SHARE-THIS.--Sharing-this-will-allow-someone-to-log-in-as-you-and-to-steal-your-ROBUX-and-items.|_D4EAD3B0A67BBEB83D23CB0A7C9E8F9ED920B887D5E8D623441C940D2120CFF908E00A1FBFDFCCE5E7B7C9A459ED20C4CAD8D80C09ED16A7954FCACC6E086D1957B539BF373CAE4F064FDEABA4C008963158C82CD82A5E5467C962EDF1AACFC29A4B33837B64F9B0FC34E32B92172AD6138AE7FBF48EC23C0F9A996F134C94EB03FB1A9A9A0C9E503C7358FEDDBA22F978C96294B66E285277497F1FA285D2DA0C87B6982B62535EDBD431E3DEAC41C86340C80AD9EB0745688E50C3AE92A74D28F02918A4524D977F0D64A26EB955B3ED4E0914101C970FCD3BB3528ACFEC57C0A890C98A927BA6406F363F60F4C6EFDF2935EBE05939A7391FA56662FD8E61DA2C665000F6FCEEAF637AFBCE23A70D3F9973F2A049670F9EE20D918553B62DEC179DA691061B4F1A8A2CF9933EAF69915537F03CF621F41CD283752134A3611A80E414328ACDE7A5792E96686813C1907E989DDBD2ED5CE335AF7DE33B51B6FF5EE348DE32F99BF2C8BE7507870F2BAF838F8A' }}
+      - lune
 
 jobs:
   unit-tests:
     name: Unit Tests
     runs-on: windows-latest
-    timeout-minutes: 30
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v2.3.4
 
       - name: Install Foreman
-        uses: rojo-rbx/setup-foreman@v1.0.1
+        uses: Roblox/setup-foreman@v2.1
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Install NPM Dependencies
         uses: bahmutov/npm-install@v1.7.4
 
-      - name: Compile Source Files
-        run: npm run build
-
-      - name: Install Latest Test Types
+      - name: Setup Tests
         run: npm run test-setup
```

Comment:
It's counter-intuitive that `npm test` / `npm run test` would execute all the `test-*` tasks _except_ setup. I would suggest `test-setup` is renamed to not have the `test` prefix, but instead something like `update-tests-types`.